scripts/list_hwm.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+from pathlib import Path, PurePath
+import pykwalify.core
+import sys
+import yaml
+
+SOC_METADATA_SCHEMA = '''
+## A pykwalify schema for basic validation of the structure of a
+## metadata YAML file.
+##
+# The socs.yml file is a simple list of key value pairs containing SoCs
+# and their location which is used by the build system.
+type: map
+mapping:
+  socs:
+    required: true
+    type: seq
+    sequence:
+      - type: map
+        mapping:
+          name:
+            required: false
+            type: str
+            desc: Name of the SoC
+          series:
+            required: false
+            type: str
+            desc: SoC series of the SoC.
+          family:
+            required: false
+            type: str
+            desc: SoC family of the SoC.
+          folder:
+            required: true
+            type: str
+            desc: Location of the SoC implementation relative to the socs.yml file.
+          vendor:
+            required: false
+            type: str
+            desc: SoC series of the SoC.
+                  This field is of informational use and can be used for filtering of SoCs.
+          arch:
+            required: false
+            type: str
+            desc: architecture of the SoC on a single-core SoC, for a multi-core SoC with
+                  mixed architectures, the value mixed can be used.
+                  This field is of informational use and can be used for filtering of SoCs.
```

Comment:
Likewise with this, a vendor should know what arch(es) their SoC is when submitting it

Reply:
the field is of informational use, and is intended to allow users the ability of doing a `list_hwm --arch=arm`, but it is not required or used by the build system itself, therefore optional.

But if Zephyr wants it required for consistency purposes then I will be happy to make it required.
Just remember, as hwm v2 remove the arch in folder path, then that means a vendor's SoC implementaion for several archs can now be grouped in a single tree, and thus arch field becomes less meainingful.
Consider a SoC folder containing SoCs of 4 different architectures.